### PR TITLE
enable running asyncio

### PR DESCRIPTION
**Pull request by dhalbert**
These are some small changes needed to use the MicroPython `asyncio` Python code. Simple task creation and running now work, using a revised version of MicroPython's code, found here: <https://github.com/adafruit/Adafruit_CircuitPython_asyncio>

I would like to get this in to 7.1.x, for experimental purposes, so people can try it, without promising that it will remain the same as it evolves.

### CODE REVIEW

**Review by jepler — APPROVED** [review]
This looks nice & simple.  I take your word for it about needing to switch to -Os for size reasons, though it's a slight bummer.  The MICROPY_EVENT_POLL_HOOK -> RUN_BACKGROUND_TASKS change makes it make sense whyI just saw select as locking up during my testing with my own branch, I think.

File: ports/stm/boards/pyb_nano_v2/mpconfigboard.mk [hunk 1]
```diff
@@ -16,9 +16,9 @@ LD_FILE = boards/STM32F411_fs.ld
 # Too big for the flash
 CIRCUITPY_AUDIOCORE = 0
 CIRCUITPY_AUDIOPWMIO = 0
-CIRCUITPY_KEYPAD = 0
-CIRCUITPY_GIFIO = 0
-CIRCUITPY_MIDI = 0
-CIRCUITPY_MSGPACK = 0
```
Comment (jepler) [inline ports/stm/boards/pyb_nano_v2/mpconfigboard.mk:22]: did midi get turned back _on_ by this change?

**Review by jepler — COMMENTED** [review]
fine to merge if it passes, though one of the config changes may not have been what you intended.

File: ports/stm/boards/pyb_nano_v2/mpconfigboard.mk (hunk 1 above)
Comment (dhalbert) [reply ports/stm/boards/pyb_nano_v2/mpconfigboard.mk:22]: There is no `CIRCUITPY_MIDI`, or at least not anymore (I can't remember if that's what it used to be) . It is `CIRCUITPY_USB_MIDI`. I left it on for this general-purpose board and turned some other stuff off.

**Comment by dhalbert** [comment]
@jepler no more build overflows; ready for re-review; thanks

**Review by jepler — APPROVED** [review]
:+1:

**Review by tannewt — COMMENTED** [review]
FWIW `poll.ipoll()` is not in CPython. https://docs.python.org/3/library/select.html#polling-objects

**Comment by dhalbert** [comment]
> FWIW `poll.ipoll()` is not in CPython. https://docs.python.org/3/library/select.html#polling-objects

Thanks, I will take that out. This was just to get things to work. I haven't tested `select` in any detail yet; my tests so far have not used it.

**Pull request merged**